src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramParser.java

```diff
@@ -46,6 +46,12 @@
     static final ParseField OFFSET = new ParseField("offset");
     static final ParseField PRE_OFFSET = new ParseField("", "pre_offset");
     static final ParseField POST_OFFSET = new ParseField("", "post_offset");
+    static final ParseField PRE_ZONE = new ParseField("", "pre_zone");
+    static final ParseField POST_ZONE = new ParseField("", "post_zone");
```

Comment:
Yes, for that case makes sense. But for options like in the above case `post_zone` which is dropped entirely, wouldn't then be empty argument to withAllDeprecated() be the right thing? Since there is no other option replacing it.